neqo-crypto/src/aead.rs

```diff
@@ -94,15 +128,26 @@ impl Aead {
             ctx: AeadContext::from_ptr(ctx)?,
         })
     }
+}
 
-    /// Decrypt a plaintext.
-    ///
-    /// The space provided in `output` needs to be larger than `input` by
-    /// the value provided in `Aead::expansion`.
-    ///
-    /// # Errors
-    /// If the input can't be protected or any input is too large for NSS.
-    pub fn encrypt<'a>(
+impl fmt::Debug for dyn Aead {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if self.fuzzing_mode() {
+            write!(f, "[FUZZING AEAD]")
+        } else {
+            write!(f, "[AEAD Context]")
+        }
+    }
+}
+
+impl Aead for AeadImpl {
+    #[must_use]
+    #[allow(clippy::unused_self)]
```

Comment:
Is this warning needed for a trait implementation?

Reply:
We'll go with non-trait version, so this comment will be ignored.